src/cells.cpp

```
@@ -225,69 +114,39 @@ void SmithLife::compute(Board &board) {
 void BasicLife::compute(Board *board) {
-    std::vector<std::vector<DiscreteAutomaton*>> grid = board->get_grid();
+    std::vector<std::vector<std::unique_ptr<DiscreteAutomaton>>>& grid = board->get_grid();
     for (int i = 0; i < board->get_height(); i++) {
         for (int j = 0; j < board->get_width(); j++) {
             BasicLife *cell = dynamic_cast<BasicLife*>(board->get_cell(i, j));
             if (cell) {
-                double new_value = cell->get_value() + 1;
-                cell = new BasicLife(i, j, new_value);
-                grid[i][j] = cell;
+                double new_value = cell->get_value()+1;
+                grid[i][j] = std::make_unique<BasicLife>(i, j, new_value);
```

cppcheck (2.10-2)

[error] Using reference to dangling temporary.

(danglingTempReference)